astropy/coordinates/representation.py

```diff
@@ -1832,15 +1832,29 @@ def _combine_operation(self, op, other, reverse=False):
     def transform(self, matrix):
         """Radial representations cannot be transformed by a Cartesian matrix.
 
+        Parameters
+        ----------
+        matrix : array-like
+            The transformation matrix in a Cartesian basis.
+            Must be a multiplication: a diagonal matrix with identical elements.
+            Must have shape (..., 3, 3), where the last 2 indices are for the
+            matrix on each other axis. Make sure that the matrix shape is
+            compatible with the shape of this representation.
+
         Raises
         ------
         NotImplementedError
+            If the matrix is not a multiplication.
 
         """
-        raise NotImplementedError(
-            "Radial representations cannot be transformed "
-            "by a Cartesian matrix."
-        )
+        scl = matrix[..., 0, 0]
+        # check that the matrix is a scaled identity matrix on the last 2 axes.
+        if np.any(matrix != scl[..., np.newaxis, np.newaxis] * np.identity(3)):
+            raise NotImplementedError("Radial representations can only be "
```

Comment:
This probably should just be a `ValueError` - it is not like we could implemented it if we wanted to!

Reply:
✅